src/vehicle.cpp

```diff
@@ -4762,6 +4882,9 @@ void vehicle::refresh()
     alternator_load = 0;
     camera_epower = 0;
     has_atomic_lights = false;
+    plow_engine_drag = 0;
```

Comment:
Your current solution has 3 different variables that are summed anyway. Just merge them into a single variable like `extra_part_drag`. Just the drag calculations.
Then add a flag for parts that cause drag, so that you aren't checking `has_flag("PLOW") || has_flag("REAPER") || has_flag("PLANTER")`, just `has_flag("POWERED_PART")` for the extra drag calculations. You'd still check for plow/reaper/planter in their exclusive sections (ie. actual plowing etc.).

Though I'd name the flag differently. POWERED_PART be interpreted as electrical power, DRIVETRAIN_POWERED is unambiguous. Or EXTRA_DRAG.